File: TravelService/Models/Destination.cs
using System;

namespace TravelService.Models
{
    public class Destination
    {
        public int Id { get; set; }

        public string Name { get; set; } = string.Empty;
        public string Location { get; set; } = string.Empty;

        public DateTime ArrivalDate { get; set; }
        public DateTime DepartureDate { get; set; }

        public string Description { get; set; } = string.Empty;
        public string Notes { get; set; } = string.Empty;

        public int TravelPlanId { get; set; }
        public TravelPlan TravelPlan { get; set; } = null!;
    }
}

File: travel-planner-frontend/src/models/Destination.js
export const createDestinationModel = (data = {}) => ({
    name: data.name || '',
    location: data.location || '',
    arrivalDate: data.arrivalDate || '',
    departureDate: data.departureDate || '',
    description: data.description || '',
    notes: data.notes || '',
});